File: cpp/1/Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ret;
        unordered_map<int, int> um;
        for(int i=0;i<nums.size();++i){
            if(um.find(nums[i])!=um.end()){
                ret.push_back(um[nums[i]]);
                ret.push_back(i);
            }
            else{
                um[target-nums[i]] = i;
            }
        }
        return ret;
    }
};

File: javascript/1/Two_Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    map = {};
    ret = [];
    var i = 0;
    while(i<nums.length){
        if(map[nums[i]]){
            ret.push(map[nums[i]]);
            ret.push(i+1);
            break;
        }
        else{
            map[target-nums[i]] = i+1;
        }
        i++;
    }
    return ret;
};